src/components/RadioButton/RadioButton.jsx

```diff
@@ -1,51 +1,55 @@
-import { uniqueId } from 'lodash';
-import React from 'react';
-import PropTypes from 'prop-types';
-import cx from 'classnames';
-import {
-  getDataAttrs,
-  getDataProps,
-} from '../../dataUtils';
-import style from './style.scss';
+import { uniqueId } from "lodash";
+import React from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import { getDataAttrs, getDataProps } from "../../dataUtils";
+import style from "./style.scss";
 
 class RadioButton extends React.Component {
   render() {
-    const id = uniqueId('id');
+    const id = uniqueId("id");
     const wrapperClasses = cx(style.wrapper, {
       [style.isLarge]: this.props.isLarge,
+      [style.disabled]: this.props.disabled
     });
 
     return (
-      <div
-        {...getDataAttrs(this.props.data)}
-      >
+      <div {...getDataAttrs(this.props.data)}>
         <label htmlFor={id} className={wrapperClasses}>
           <input
-            className={style.original}
+            className={cx(style.original, {
+              [style.disabled]: this.props.disabled
+            })}
             type="radio"
             id={id}
             name={this.props.name}
             onChange={this.props.onChange}
-            checked={this.props.checked}
+            checked={
+              this.props.checked || this.props.value === this.props.fieldValue
+            }
+            value={this.props.fieldValue}
```

Comment:
added "fieldValue" prop so we can give the radiobutton a group in a radio group scenario